lib/fizzy/parser.hpp

```diff
@@ -38,12 +38,13 @@ inline parser_result<uint8_t> parse_byte(const uint8_t* pos, const uint8_t* end)
 /// Parse `expr`, i.e. a function's instructions residing in the code section.
 /// https://webassembly.github.io/spec/core/binary/instructions.html#binary-expr
 ///
-/// @param input    The beginning of the expr binary input.
-/// @param end      The end of the binary input.
-/// @param func_idx Index of the function being parsed.
-/// @param module   Module that this code is part of.
-parser_result<Code> parse_expr(
-    const uint8_t* input, const uint8_t* end, FuncIdx func_idx, const Module& module);
+/// @param input       The beginning of the expr binary input.
+/// @param end         The end of the binary input.
+/// @param local_count The number of locals defined.
```

Comment:
We are going to need types for type validation, maybe makes sense to pass `const Locals&` right away?

Reply:
This was discussed in the an earlier comment. I started out with `std::vector<Locals>` but then local counting and validation of it has to be moved into `parse_expr`, which I think isn't the nicest.

I'd defer this work once we implement type checking.

Reply:
Maybe create `Code` object in `Module` before `parse_expr`?

Reply:
> Maybe create `Code` object in `Module` before `parse_expr`?

I would advice against this, because it would make `code` confusingly both input and output parameter

Reply:
It does not mean that `parse_expr` should return `Code`. Maybe it should return `Expr`. But is also bad to ask `parse_expr` to move the information about locals (here `local_count`, maybe type information later) from one place to another. That's unneeded and trivial responsibility.